masonry_core/src/properties/types/gradient.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::kurbo::{Point, Rect};
+use crate::peniko::color::{ColorSpaceTag, HueDirection};
+use crate::peniko::{ColorStops, ColorStopsSource, Extend};
+
+#[derive(Clone, Debug)]
+pub enum GradientShape {
+    /// Gradient that transitions between two or more colors along a line.
+    Linear {
+        /// The angle defining the gradient line's direction.
+        /// Zero degrees points upwards and positive angles represent clockwise rotation.
+        angle: f64,
+    },
+}
+
+#[derive(Clone, Debug)]
+pub struct Gradient {
```

Comment:
This should have docs (we should also mention Peniko's gradient type here, including it's lack of translation invariance as the reason for this being a separate type)

At the very least, the docs should mention that this is a widget property (maybe also what built-in widgets make use of it, but that's more questionable).

Reply:
Ha, indeed; exactly my point - this isn't a widget property, but I thought it was. That's what docs are for - this should link up to the `Background` property.

Reply:
This should hopefully be clearer once the Property trait is merged.

The general idea is you have a module with properties, and you have a sub-module with values used *in* properties that aren't properties themselves.

Yes, I should probably document that.